### DISCUSSION TITLE
What do you think about an Objeck backend for Swig?

### DISCUSSION THREAD

**Discussion by ghost** [objeck/objeck-lang#143] [Ideas]
Instead of doing your own binding generator that needs to be adapted for each libraries:

https://www.swig.org/Doc4.1/Extending.html

**Comment by objeck** [comment]
The QML model seems similar to the client/server HTTP model. 

Meaning UI and calls to the backend are architecturally separated. The backend does not need to know what controls sent the data; just process the data sent, and formulate a response. This model seems clean, but I still need to play around with the code.

**Reply by ghost** [reply]
What does Swig has anything to do with this?

**Reply by ghost** [reply]
Swig is a binding generator. It parsed the C/C++ headers for you. It's your job to use the parsed data to generate binding for the target language (aka adding new backend). It will help with your ML adventure (Swig will parse the C/C++ headers much better than your own parser).
